velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -496,6 +496,26 @@ struct DayOfYearFunction {
   }
 };
 
+template <typename T>
+struct WeekdayFunction : public InitSessionTimezone<T> {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  // 0 = Monday, 1 = Tuesday, ..., 6 = Sunday
+  FOLLY_ALWAYS_INLINE int32_t getWeekday(const std::tm& time) {
+    return (time.tm_wday - 1 + 7) % 7;
+  }
+
+  FOLLY_ALWAYS_INLINE void call(
+      int32_t& result,
+      const arg_type<Timestamp>& timestamp) {
```

Comment:
According to Spark code (see [link](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala#L460)), `weekday` function doesn't directly accept Timestamp input, but only accepts Date input. For Timestamp input, Spark will plan a cast expression to convert it to Date type. So I think we only need consider Timestamp type in Velox.

Reply:
I see that `dayofweek` also extends `GetDateField` but has been registered for `Date` and `Timestamp` in Velox. Should that be updated to just `Date` as well? https://github.com/facebookincubator/velox/blob/7b68a82638fa1a27ecb5cf4d3f82100481a812da/velox/functions/sparksql/Register.cpp#L304

Reply:
Yes, please go ahead to update it.

Reply:
Updated both functions, thanks!